modules/core/arrow-generic/src/main/kotlin/arrow/generic/Generic.kt

```diff
@@ -0,0 +1,10 @@
+package arrow.generic
+
+interface Generic<T, Repr> {
```

Comment:
As per the signatures, this interface models any isomorphism, not just generic representation. What about renaming it to `Cast` or `Transform` or `As`.

Reply:
I'm not sure any of those names convey the fact that is bidirectional and preserves structure. In any case `Iso` would be a better name but we already have an Iso in Optics. This is not yet in use and I may discard it, I'll move it to type classes if we end up using it for the generic derivation of arbitrary type classes.